whois-api/src/main/java/net/ripe/db/whois/api/httpserver/CaseInsensitiveRewriteRegexRule.java

```diff
@@ -1,19 +1,28 @@
 package net.ripe.db.whois.api.httpserver;
 
 import org.eclipse.jetty.rewrite.handler.RewriteRegexRule;
+import org.eclipse.jetty.rewrite.handler.Rule;
 
+import java.io.IOException;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class CaseInsensitiveRewriteRegexRule extends RewriteRegexRule {
 
+    private final String regex;
+
     public CaseInsensitiveRewriteRegexRule(String regex, String replacement) {
-        super(regex, replacement);
+        this.regex = regex;
+        setReplacement(replacement);
         setTerminating(true);
     }
 
     @Override
-    public void setRegex(String regex) {
-        _regex = regex == null ? null : Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+    public Rule.Handler matchAndApply(Rule.Handler input) throws IOException {
+        String target = this.isMatchQuery() ? input.getHttpURI().getPathQuery() : input.getHttpURI().getPath();
+        Matcher matcher =  Pattern.compile(regex, Pattern.CASE_INSENSITIVE).matcher(target);
+        return matcher.matches() ? this.apply(input, matcher) : null;
```

Comment:
Add Nullable annotation, what happens if null is returned?

Reply:
I just override the default one to add Pattern.CASE_INSENSITIVE. RegexRule itself sends the null incase it does not match

Reply:
OK